packages/medusa/src/api/store/products/middlewares.ts

```diff
@@ -23,6 +26,35 @@ export const storeProductRoutesMiddlewares: MiddlewareRoute[] = [
         QueryConfig.listProductQueryConfig
       ),
       filterByValidSalesChannels(),
+      setContext({
+        with_inventory_quantity: async (req: MedusaRequest) => {
+          const withInventoryQuantity = req.remoteQueryConfig.fields.some(
+            (field) => field.includes("variants.inventory_quantity")
+          )
+
+          req.remoteQueryConfig.fields = req.remoteQueryConfig.fields.filter(
+            (field) => !field.includes("variants.inventory_quantity")
+          )
+
+          return withInventoryQuantity
+        },
```

Comment:
suggestion(non-blocking): We could kill this dedicated context property, move the logic here to `stock_location_id`, and check the `remoteQueryConfig.fields` in the route? Just to have fewer magic props

Reply:
Yup I agree with this approach as well